.github/push-image.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+set -e
+
+if [ "$#" -ne 1 ]; then
+    echo "Pushes a container image to Docker Hub with tags"
+    echo 
+    echo "Usage: $0 project"
+    echo "Example: $0 dummy-api"
+    echo "Relies on the following environment variables:"
+    echo "- GITHUB_REF, GITHUB_SHA (GH Action default)"
+    echo "- DOCKER_USERNAME, DOCKER_PASSWORD"
+    echo "Uses project:latest as input image."
+    echo "Override this behaviour with IMAGE_TAG environment variable."
+    exit 1
+fi
+
+project="$1"
+IMAGE_TAG="${IMAGE_TAG:-latest}"
+
+
+# login
+DOCKER_REGISTRY="${DOCKER_REGISTRY:-}"
+echo "${DOCKER_PASSWORD}" | docker login --username "${DOCKER_USERNAME}" --password-stdin "${DOCKER_REGISTRY}"
+
+# tag temporarily as liberoadmin due to lack of `libero/` availability
+name="${DOCKER_REGISTRY}liberoadmin/${project}"
+input_image="libero/${project}:${IMAGE_TAG}"
+
+if [[ "$GITHUB_REF" == "refs/heads"* ]]; then
+    # push `branch-sha` tagged image
+    branch="${GITHUB_REF/refs\/heads\//}"
+    docker tag "$input_image" "${name}:${branch}-${GITHUB_SHA}"
+    docker push "${name}:${branch}-${GITHUB_SHA}"
+
+    if [[ "$branch" = "master" ]]; then
+        # push `latest` tag
+        docker tag "$input_image" "${name}:latest"
+        docker push "${name}:latest"
+    fi
+
+elif [[ "$GITHUB_REF" =~ refs/tags/v(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*) ]]; then
```

Comment:
Do I understand correctly that if "1.2.3" is tagged after "1.2.4" then the version in docker for "1.2" would actually be "1.2.3"?

Reply:
Yes. Do you think this an issue we should address with automated checks? And if so where?

My gut tells me that checks for this should reside with the github repo and not rely on a central libero/script that compares the git tag to existing container images.

I think the editor team actually automates the tagging of the repo with husky. Relying on [convetional commits](https://conventionalcommits.org).

There are also github actions available that perform tagging based on issue labels and hashtags inside PR messages: https://github.com/phish108/autotag-action

Reply:
I agree on reflection... I'll close this out, but its probably worth noting to explain to ppl this is how it works :+1: 

> My gut tells me that checks for this should reside with the github repo and not rely on a central libero/script that compares the git tag to existing container images.

Reply:
Having tried to manage version numbers manually in the past, I never want to have to deal with them again. For Editor, at the moment I have setup 'standard-release' and tied it to the npm script 'release'. This, coupled with using husky to (sort of) enforce that commit messages adhear to the 'conventional commits' standard means that it bumps version numbers and creates tags for me.
Id originally wanted to manage it server side, with GitHub releases and git hooks, but you need an enterprise sub for that sadly.